dds/DCPS/transport/shmem/ShmemTransport.cpp

```diff
@@ -286,7 +309,7 @@ ShmemTransport::ReadTask::signal_semaphore()
   ACE_OS::sema_post(&semaphore_);
 }
 
-void
+bool
```

Comment:
Is the return value used?

Reply:
It used in the inner loop in of `ReadTask::svc`.

Reply:
It is... but I'm not sure this is necessary.  

In the original design, `read_from_links` happens once per successful wait on the semaphore (since it's a counting semaphore, it knows how many "resources" are available).  Would that work now that the rest of the transport framework integration is done?

If the answer to the above ends up being "no," then could `read_from_links` just do all the work it needs to do before returning?  I guess the downside there is potentially starving-out the `stopped_` signal, but that could be passed by reference if needed.

One reason I brought this up is that if we keep it in we'll need to review each path through `read_from_links` to see if it's returning the correct value.